File: core/src/com/alexpi/awesometanks/entities/projectiles/Flame.java
package com.alexpi.awesometanks.entities.projectiles;

import com.badlogic.gdx.assets.AssetManager;
import com.badlogic.gdx.graphics.g2d.Batch;
import com.badlogic.gdx.math.Vector2;
import com.badlogic.gdx.physics.box2d.CircleShape;
import com.badlogic.gdx.physics.box2d.World;
import com.badlogic.gdx.scenes.scene2d.Group;
import com.alexpi.awesometanks.entities.actors.ParticleActor;

/**
 * Created by Alex on 16/01/2016.
 */
public class Flame extends Projectile {

    public float burnDuration;
    private final ParticleActor particleActor;

    public Flame(AssetManager manager, World world, Vector2 pos, float angle, float burnDuration, boolean filter) {
        super(world, pos, new CircleShape(), angle, 15f, .1f, 20f, filter);
        this.burnDuration = burnDuration;
        particleActor = new ParticleActor(manager,"particles/flame.party",getX()+getWidth()/2,getY()+getHeight()/2,true);
    }

    @Override
    public void draw(Batch batch, float parentAlpha) {
        super.draw(batch, parentAlpha);
        particleActor.draw(batch, parentAlpha);
    }

    @Override
    public void act(float delta) {
        super.act(delta);
        particleActor.setPosition(getX() + getWidth() / 2, getY()+getHeight()/2);
        particleActor.act(delta);
    }

}

File: core/src/com/alexpi/awesometanks/entities/projectiles/Flame.kt
package com.alexpi.awesometanks.entities.projectiles

import com.alexpi.awesometanks.entities.actors.HealthOwner
import com.alexpi.awesometanks.entities.actors.ParticleActor
import com.alexpi.awesometanks.entities.components.body.BodyShape
import com.badlogic.gdx.graphics.g2d.Batch
import com.badlogic.gdx.math.Vector2
import com.badlogic.gdx.scenes.scene2d.Actor

/**
 * Created by Alex on 16/01/2016.
 */
class Flame(pos: Vector2, angle: Float, private var burnDuration: Float, filter: Boolean) : Projectile(
    pos, BodyShape.Circular(.05f), angle, 15f, 20f, filter
) {
    private val particleActor: ParticleActor = ParticleActor(
        "particles/flame.party",
        x + bodyShape.width / 2,
        y + bodyShape.height / 2,
        true
    )

    override fun handleCollision(actor: Actor) {
        super.handleCollision(actor)
        if(actor is HealthOwner) actor.healthComponent.burn(burnDuration, .35f)
    }

    override fun draw(batch: Batch, parentAlpha: Float) {
        super.draw(batch, parentAlpha)
        particleActor.draw(batch, parentAlpha)
    }

    override fun act(delta: Float) {
        super.act(delta)
        particleActor.setPosition(x + bodyShape.width / 2, y + bodyShape.height / 2)
        particleActor.act(delta)
    }

}